bootcompiler/src/main/scala/org/mozartoz/bootcompiler/transform/Desugar.scala

```diff
@@ -187,6 +187,25 @@ object Desugar extends Transformer with TreeDSL {
       treeCopy.MatchStatement(expr, value,
           clauses map(c => matchExpressionClauseToBindStatement(result, c)),
           exprToBindStatement(result, elseExpression))
+
+    case Record(label, fields) =>
+      val lastI = fields.lastIndexWhere(_.value match {
+        case CallExpression(callable, args) => true
+        case _                              => false
+      })
```

Comment:
Indeed, correctness is the most important but it's not clear what is correct.
At least, values should be evaluated in order except for the tail call.

Doing the trick to move the record construction before the call of one of the values could be observed with `IsDet` for instance, as it allows to observe the partially built record.
In pure dataflow `IsDet` is not allowed but it would still be possible to see it with any side effect like the Browser whether the value is computed as soon as the record is built or deferred.
I think this is currently acceptable for the last element if it is a tail call.
We could technically support a nested tail call as long as it is tail, but I'm not sure it's worth the effort and it might be confusing to the user.